### DISCUSSION TITLE
How to handle out of order / late events in processors

### DISCUSSION THREAD

**Discussion by flomickl** [apache/streampipes#1301] [General]
Just a technical question.

How does StreamPipes handle out of order / late events?
Is it implemented in some way, so it can be handled in the processors?
I mean window functions or other micro batch processors.

IoT devices are most of the time in the internet ^^ but the effect still can occur

**Comment by tenthe** [comment]
Hi @flomickl,
this is something that the processing element developer needs to address in the implementation. So far there is no standard funcionality. But this would be good to add.
Cheers,

Philipp
